query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PrometheusBasedRequestHandler.java

```diff
@@ -3,30 +3,56 @@
 import com.google.common.base.Preconditions;
 import com.typesafe.config.Config;
 import io.reactivex.rxjava3.core.Observable;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
 import java.util.Optional;
+import okhttp3.Request;
 import org.hypertrace.core.query.service.ExecutionContext;
 import org.hypertrace.core.query.service.QueryCost;
+import org.hypertrace.core.query.service.QueryRequestUtil;
 import org.hypertrace.core.query.service.RequestHandler;
+import org.hypertrace.core.query.service.api.Expression;
+import org.hypertrace.core.query.service.api.Expression.ValueCase;
 import org.hypertrace.core.query.service.api.QueryRequest;
 import org.hypertrace.core.query.service.api.Row;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class PrometheusBasedRequestHandler implements RequestHandler {
 
+  private static final Logger LOG = LoggerFactory.getLogger(PrometheusBasedRequestHandler.class);
+
   private static final String VIEW_DEFINITION_CONFIG_KEY = "prometheusViewDefinition";
   private static final String TENANT_ATTRIBUTE_NAME_CONFIG_KEY = "tenantAttributeName";
   private static final String START_TIME_ATTRIBUTE_NAME_CONFIG_KEY = "startTimeAttributeName";
 
   private final QueryRequestEligibilityValidator queryRequestEligibilityValidator;
   private final String name;
-  private PrometheusViewDefinition prometheusViewDefinition;
+  private final QueryRequestToPromqlConverter requestToPromqlConverter;
+  private final PrometheusRestClient prometheusRestClient;
+
   private Optional<String> startTimeAttributeName;
-  private QueryRequestToPromqlConverter requestToPromqlConverter;
+  private PrometheusViewDefinition prometheusViewDefinition;
 
   PrometheusBasedRequestHandler(String name, Config config) {
     this.name = name;
     this.processConfig(config);
     this.queryRequestEligibilityValidator =
         new QueryRequestEligibilityValidator(prometheusViewDefinition);
+    this.requestToPromqlConverter = new QueryRequestToPromqlConverter(prometheusViewDefinition);
+    this.prometheusRestClient = PrometheusRestClientFactory.get().getPrometheusClient(name);
+  }
+
+  PrometheusBasedRequestHandler(
+      String name, Config config, PrometheusRestClient prometheusRestClient) {
```

Comment:
Is this requestHanlder config? If, yes, can we rename to `requestHandlerConfig`?